modules/reactor/src/meshgenerators/PeripheralTriangleMeshGenerator.C

```diff
@@ -78,74 +84,54 @@ PeripheralTriangleMeshGenerator::validParams()
 
 PeripheralTriangleMeshGenerator::PeripheralTriangleMeshGenerator(const InputParameters & parameters)
   : MeshGenerator(parameters),
-    _input_name(getParam<MeshGeneratorName>("input")),
+    _input_names(getParam<std::vector<MeshGeneratorName>>("input")),
     _peripheral_ring_radius(getParam<Real>("peripheral_ring_radius")),
     _peripheral_ring_num_segments(getParam<unsigned int>("peripheral_ring_num_segments")),
     _desired_area(getParam<Real>("desired_area")),
-    _desired_area_func(getParam<std::string>("desired_area_func")),
-    _peripheral_ring_block_name(isParamValid("peripheral_ring_block_name")
-                                    ? getParam<SubdomainName>("peripheral_ring_block_name")
-                                    : (SubdomainName) ""),
-    _external_boundary_name(getParam<BoundaryName>("external_boundary_name"))
+    _desired_area_func(getParam<std::string>("desired_area_func"))
 {
-  // Calculate outer boundary points
+  declareMeshesForSub("input");
 
-  std::vector<libMesh::Point> outer_polyline;
-  // radial spacing
-  Real d_theta = 2.0 * M_PI / _peripheral_ring_num_segments;
-  for (unsigned int i = 0; i < _peripheral_ring_num_segments; i++)
+  auto params = _app.getFactory().getValidParams("FlexiblePatternGenerator");
+  params.set<std::vector<MeshGeneratorName>>("inputs") = _input_names;
+  params.set<MooseEnum>("boundary_type") = "circle";
+  params.set<Real>("boundary_size") = _peripheral_ring_radius * 2;
+  params.set<unsigned int>("boundary_sectors") = _peripheral_ring_num_segments;
+  auto & pos = params.set<std::vector<Point>>("extra_positions");
+  auto & ids = params.set<std::vector<unsigned int>>("extra_positions_mg_indices");
+  for (const auto i : index_range(_input_names))
   {
```

Comment:
```suggestion
  params.set<std::vector<Point>>("extra_positions") = std::vector<Point>(_input_names,size(), Point());
  auto & ids = params.set<std::vector<unsigned int>>("extra_positions_mg_indices");
  ids.resize(_input_names,size());
  std::iota(ids.begin(), ids.end(),0);
```

Reply:
Changed `extra_positions` but kept `extra_positions_mg_indices` because the new way has the same three lines with more letters.

Reply:
If we are as serious as counting lines, `push_back` allocates memory every time it is called.

Reply:
>every time it is called

Not true. I do not know how much it allocates when hitting the limit, but I am sure it is not allocating one more entry.